### DISCUSSION TITLE
Copy & Paste Fieldsets across different pages

### DISCUSSION THREAD

**Discussion by laracwi** [statamic/cms#8572] [General]
Hi everyone, 

Is there a way to copy Fieldsets across different pages within Statamic? I've looked through the docs but couldn't find anything related. I know that I can copy individual pages of my page collection but I can only move and duplicate fieldsets on the specific page.

Thanks in advance,
Lara

**Comment by jasonvarga** [comment]
Can you show a screenshot of what you're wanting to copy? I'm not really understanding the question.

**Reply by paul-hph** [reply]
Let me jump in here to clarify.  
In other CMS it is possible to copy or move content blocks from one page to another, including the content.
Right now, i can't find an option to do that in statamic. 

We do have the option "Duplicate Set" for Field-Sets: 

![image](https://github.com/statamic/cms/assets/3603487/33f7fd5c-a78d-4656-9520-3f5e3be5da91)


This makes it possible to duplicate and reuse field sets and the content on the same page. 



**Really** helpful would be an option like "Copy Field-Set", and "Paste Field-Set" that would work across pages. 


![image](https://github.com/statamic/cms/assets/3603487/d977bf18-03eb-4865-b279-8d235d8ced01)

on another Page:

![image](https://github.com/statamic/cms/assets/3603487/23b8858c-293b-4fc9-b1d5-8ae93b44cdf9)



--- 

**To Really make this feature complete. There could be a clipboard functionality with the following options:** 

- Add Fieldset to Clipboard. 
- Paste Copy of Fieldset from Clipboard here
- Move Fieldset from Clipboard here

Allow to Copy and Paste multiple items from clipboard. 


---
Here is an Example how it works in the CMS TYPO3

**Simple copy/paste across pages:** 

https://github.com/statamic/cms/assets/3603487/e5843740-1b92-4629-8ecf-6c2371f31e7b


**Copy/Move/Paste across pages:**


https://github.com/statamic/cms/assets/3603487/c552eb04-5781-4e9d-a676-db2c5d859162

**Reply by jasonvarga** [reply]
That's a pretty cool feature. Thanks for the info.

**Comment by duncanmcclean** [comment]
Not sure if this is exactly what you're looking for but I think we did something similar recently.

We're using a page builder on every page and the client wanted to re-use certain blocks on multiple places in the site and have a single source of truth for its content.

I created a new "Shared Components" collection - which just had a title (for the description of the component) and our content builder field (which is a fieldset in itself so can be re-used in multiple collections).

Then, in my page builder, I made a new block for "Shared Component" that lets you select or create a new shared component in that collection. My partial for that block then runs through any of the configured page builder blocks in that "shared component" entry.

**Reply by paul-hph** [reply]
That's actually also a very useful feature we are using at our agency, other CMSes all the time. 
There it's called "Referenced Content Elements" It's a special kind of object that can be placed anywhere and can pull in any content element placed on any page. Although, we always recommend collecting all Content Elements that should be referenced on one page. Otherwise, i might happen that somebody removes an element on a page without knowing it was used as a reference somewhere else.

We will definitely try to replicate what you described in statamic. 

But the Copy/Paste/Move/Clipboard feature request would mainly be used for two reasons: 

1. To quickly pick and choose existing elements and copy them to build a new page and then change the content. In Many cases, this is way faster than recreating all the Content-Blocks/Fieldsets.
2. To Move single or multiple Content-Blocks/Fieldsets to other pages.

**Comment by DennisMeene** [comment]
Hey, this feature would be pretty dope! Especially for more complex content elements it would be nice to be able to copy-paste finished elements to other pages without the need to do it via editing the YAML files. Are there any plans to implement it in a future update?

**Reply by robdekort** [reply]
There's even a PR by the wonderful @jacksleight: https://github.com/statamic/cms/pull/10359. His field actions PR already got merged and releases so hopefully this one will follow soon.

**Reply by DennisMeene** [reply]
Oh, thanks for the info, very nice to hear!
